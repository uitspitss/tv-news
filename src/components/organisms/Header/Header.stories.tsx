import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import base from 'paths.macro';

import { Header, HeaderProps } from './Header';

const path = base.replace(
  /^.+\/(atoms|molecules|organisms|templates|pages)\/(.+)\//,
  '$1/$2',
);

export default {
  title: `${path}`,
  component: Header,
  argTypes: {
    isOpen: { control: 'boolean' },
  },
} as Meta;

const Template: Story<HeaderProps> = (args) => <Header {...args} />;

export const Desktop = Template.bind({});

export const Mobile = Template.bind({});
Mobile.parameters = {
  viewport: {
    defaultViewport: 'iphonex',
  },
};
