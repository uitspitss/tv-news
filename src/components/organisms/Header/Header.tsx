import React, { memo, VFC, useState } from 'react';
import tw, { styled } from 'twin.macro';
import Link from 'next/link';
import {
  TwitterShareButton,
  TwitterIcon,
  FacebookShareButton,
  FacebookIcon,
} from 'next-share';

import SEO from '~/next-seo.config.json';

export type HeaderProps = {
  /**
   * class for twin.macro
   */
  className?: string;
};

type Props = {
  isOpen: boolean;
  handleClick: () => void;
} & HeaderProps;

const Component: VFC<Props> = (props) => {
  const { className } = props;

  return (
    <div className={className}>
      <nav className="nav">
        <div className="title">
          <Link href="/">
            <a>{SEO.defaultTitle}</a>
          </Link>
        </div>
        <div className="share">
          <FacebookShareButton className="button" url={SEO.canonical}>
            <FacebookIcon size={24} round />
          </FacebookShareButton>
          <TwitterShareButton className="button" url={SEO.canonical}>
            <TwitterIcon size={24} round />
          </TwitterShareButton>
        </div>
      </nav>
    </div>
  );
};

const StyledComponent = styled(Component)`
  & {
    ${tw`bg-primary-700`}
    & > .nav {
      ${tw`flex items-center justify-between py-2 px-2 lg:px-6`}

      & > .title {
        ${tw`flex items-center flex-shrink-0 mr-6 text-gray-100 font-semibold text-xl tracking-tight`}
      }

      & > .share {
        ${tw`flex items-center`}

        & > :not(.button:last-child) {
          ${tw`mr-2`}
        }
      }
    }
  }
`;

/**
 * nav in header
 */
export const Header = memo((props: HeaderProps) => {
  const [isOpen, setIsOpen] = useState(false);
  const handleClick = () => {
    setIsOpen(!isOpen);
  };

  return (
    <StyledComponent {...props} isOpen={isOpen} handleClick={handleClick} />
  );
});
