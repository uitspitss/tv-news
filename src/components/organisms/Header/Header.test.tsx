import React from 'react';
import { render } from '@testing-library/react';

import { Header } from './Header';

describe('Header', () => {
  test('snapshot test', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
