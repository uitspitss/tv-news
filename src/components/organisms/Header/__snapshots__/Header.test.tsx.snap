// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot test 1`] = `
.emotion-0 {
  --tw-bg-opacity: 1;
  background-color: rgba(55, 65, 81, var(--tw-bg-opacity));
}

.emotion-0>.nav {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

@media (min-width: 1024px) {
  .emotion-0>.nav {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

.emotion-0>.nav>.title {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.5rem;
  --tw-text-opacity: 1;
  color: rgba(243, 244, 246, var(--tw-text-opacity));
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 1.75rem;
  letter-spacing: -0.025em;
}

.emotion-0>.nav>.share {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0>.nav>.share>:not(.button:last-child) {
  margin-right: 0.5rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <nav
      class="nav"
    >
      <div
        class="title"
      >
        <a
          href="/"
        >
          TV局ニュースプレイリスト
        </a>
      </div>
      <div
        class="share"
      >
        <button
          aria-label="facebook"
          class="button"
          disabledstyle="[object Object]"
          opts="[object Object]"
          style="background-color: transparent; padding: 0px; cursor: pointer;"
          url="https://tv-news.uitspitss.dev"
          windowheight="400"
          windowwidth="550"
        >
          <svg
            height="24"
            viewBox="0 0 64 64"
            width="24"
          >
            <circle
              cx="32"
              cy="32"
              fill="#3b5998"
              r="31"
            />
            <path
              d="M34.1,47V33.3h4.6l0.7-5.3h-5.3v-3.4c0-1.5,0.4-2.6,2.6-2.6l2.8,0v-4.8c-0.5-0.1-2.2-0.2-4.1-0.2 c-4.1,0-6.9,2.5-6.9,7V28H24v5.3h4.6V47H34.1z"
              fill="white"
            />
          </svg>
        </button>
        <button
          aria-label="twitter"
          class="button"
          disabledstyle="[object Object]"
          opts="[object Object]"
          style="background-color: transparent; padding: 0px; cursor: pointer;"
          url="https://tv-news.uitspitss.dev"
          windowheight="400"
          windowwidth="550"
        >
          <svg
            height="24"
            viewBox="0 0 64 64"
            width="24"
          >
            <circle
              cx="32"
              cy="32"
              fill="#00aced"
              r="31"
            />
            <path
              d="M48,22.1c-1.2,0.5-2.4,0.9-3.8,1c1.4-0.8,2.4-2.1,2.9-3.6c-1.3,0.8-2.7,1.3-4.2,1.6 C41.7,19.8,40,19,38.2,19c-3.6,0-6.6,2.9-6.6,6.6c0,0.5,0.1,1,0.2,1.5c-5.5-0.3-10.3-2.9-13.5-6.9c-0.6,1-0.9,2.1-0.9,3.3 c0,2.3,1.2,4.3,2.9,5.5c-1.1,0-2.1-0.3-3-0.8c0,0,0,0.1,0,0.1c0,3.2,2.3,5.8,5.3,6.4c-0.6,0.1-1.1,0.2-1.7,0.2c-0.4,0-0.8,0-1.2-0.1 c0.8,2.6,3.3,4.5,6.1,4.6c-2.2,1.8-5.1,2.8-8.2,2.8c-0.5,0-1.1,0-1.6-0.1c2.9,1.9,6.4,2.9,10.1,2.9c12.1,0,18.7-10,18.7-18.7 c0-0.3,0-0.6,0-0.8C46,24.5,47.1,23.4,48,22.1z"
              fill="white"
            />
          </svg>
        </button>
      </div>
    </nav>
  </div>
</div>
`;
